py3-certifi.spec

```diff
@@ -0,0 +1,40 @@
+### RPM external py3-certifi 2017.7.27.1
+## INITENV +PATH PYTHONPATH %i/${PYTHON_LIB_SITE_PACKAGES}
+
+Source: https://pypi.python.org/packages/20/d0/3f7a84b0c5b89e94abbd073a5f00c7176089f526edb056686751d5064cbd/certifi-2017.7.27.1.tar.gz
+Requires: python3
+BuildRequires: py3-setuptools
+
+%prep
+%setup -n certifi-%realversion
+
+%build
+export PYTHON3_ROOT
+python3 setup.py build
+
+%install
+mkdir -p %i/$PYTHON_LIB_SITE_PACKAGES
+export LD_LIBRARY_PATH=$PYTHON3_ROOT/lib:$LD_LIBRARY_PATH
+PYTHONPATH=%i/$PYTHON_LIB_SITE_PACKAGES:$PYTHONPATH python3 setup.py install --prefix=%i
+find %i -name '*.egg-info' -exec rm {} \;
+# replace all instances of #!/path/bin/python into proper format
+for f in `find %i -type f`; do
+    if [ -f $f ]; then
+        perl -p -i -e 's{.*}{#!/usr/bin/env python3} if $. == 1 && m{#!.*/bin/python.*}' $f
+    fi
+done
+
+# Generate dependencies-setup.{sh,csh} so init.{sh,csh} picks full environment.
+mkdir -p %i/etc/profile.d
+: > %i/etc/profile.d/dependencies-setup.sh
+: > %i/etc/profile.d/dependencies-setup.csh
+for tool in $(echo %{requiredtools} | sed -e's|\s+| |;s|^\s+||'); do
+  root=$(echo $tool | tr a-z- A-Z_)_ROOT; eval r=\$$root
+  if [ X"$r" != X ] && [ -r "$r/etc/profile.d/init.sh" ]; then
+    echo "test X\$$root != X || . $r/etc/profile.d/init.sh" >> %i/etc/profile.d/dependencies-setup.sh
+    echo "test X\$?$root = X1 || source $r/etc/profile.d/init.csh" >> %i/etc/profile.d/dependencies-setup.csh
+  fi
+done
```

Comment:
this generating dependencies-setup.*sh files section looks very commonly used. I would suggest to define it in rpm-preamble.file and then use it where needed.